controllers/taloscontrolplane_controller.go

```diff
@@ -591,6 +611,38 @@ func (r *TalosControlPlaneReconciler) bootControlPlane(ctx context.Context, clus
 	return ctrl.Result{Requeue: true}, nil
 }
 
+func (r *TalosControlPlaneReconciler) bootstrapCluster(ctx context.Context, cluster *clusterv1.Cluster, machines []clusterv1.Machine) error {
+	c, err := r.talosconfigForMachines(ctx, machines...)
+	if err != nil {
+		return err
+	}
+
+	defer c.Close() //nolint:errcheck
+
+	addresses := []string{}
+	for _, machine := range machines {
+		if len(machine.Status.Addresses) == 0 {
+			continue
+		}
+
+		for _, addr := range machine.Status.Addresses {
+			if addr.Type == clusterv1.MachineInternalIP {
+				addresses = append(addresses, addr.Address)
+
+				break
+			}
+		}
+	}
+
+	if len(addresses) == 0 {
+		return fmt.Errorf("no machine addresses to use for bootstrap")
+	}
+
+	sort.Strings(addresses)
+
+	return c.Bootstrap(talosclient.WithNodes(ctx, addresses[0]), &machineapi.BootstrapRequest{})
```

Comment:
I think we should also honor Talos error when cluster is already bootstrapped: https://github.com/talos-systems/talos/blob/e4bc68bf026966a3326872a1d342ef3b9c05cc9d/internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go#L326

This might happen for multiple reasons:

* bootstrap API was called, but CAPI controller died/never got the response - it should consider AlreadyExists as a success
* after pivoting to a new cluster, all `Status`es are lost, so control plane provider should reconcile the status for a running cluster